File: binary tree level order traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

# class Solution:
#     def levelOrderBottom(self, root: TreeNode) -> List[List[int]]:

def dfs(root, depth, result):
    if root == None:
        return
    print(depth, ":", root.val)
    
    if depth == len(result):
        result.append([])
        
    result[depth].append(root.val)
    
    dfs(root.left, depth+1, result)
    dfs(root.right, depth+1, result)

class Solution:
    def levelOrderBottom(self, root):
        result = []
        dfs(root, 0, result)
        result.reverse()
        return result

File: c++/binary_tree_level_order_traversal.cc
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
void dfs(TreeNode* root, int depth, vector<vector<int>>& result){
    if(root == nullptr){
        return;
    }
    if(depth == result.size()){
        result.push_back({});
    }
    
    result[depth].push_back(root->val);
    
    dfs(root->left, depth+1, result);
    dfs(root->right, depth+1, result);
}

class Solution {
public:
    vector<vector<int>> levelOrderBottom(TreeNode* root) {
        vector<vector<int>> result;
        dfs(root, 0, result);
        reverse(begin(result), end(result));
        return result;
    }
};